src/services/data-importer/csv/CSVDataImporter.ts

```diff
@@ -18,60 +18,54 @@
 
 import fs from 'fs-extra';
 import papaparse from 'papaparse';
+import { ObjectSchemaProperty, PropertyType } from 'realm';
 
-import { DataImporter } from '../DataImporter';
+import { DataImporter, ImportableFile } from '../DataImporter';
 
 export class CSVDataImporter extends DataImporter {
   private static readonly NUMBER_OF_INSERTS_BEFORE_COMMIT = 10000;
 
-  public import(realm: Realm) {
-    this.files.map((file, index) => {
-      const schema = this.importSchema[index];
+  public import(realm: Realm, files: ImportableFile[]) {
+    files.map((file, index) => {
+      const className = file.className;
+      const schema = realm.schema.find(s => s.name === className);
+      if (!schema) {
+        throw new Error(
+          `Unable to import ${file.path}: Class name (${className}) missing from schema`,
+        );
+      }
 
-      const rawCSV = fs.readFileSync(file, 'utf8');
+      const rawCSV = fs.readFileSync(file.path, 'utf8');
       const data = papaparse.parse<any>(rawCSV, {
         header: true, // to avoid parsing first line as data
         skipEmptyLines: true,
       }).data;
 
       realm.beginTransaction();
       let numberOfInsert = 0;
-      data.forEach((row, lineIndex: number) => {
+      for (const [rowIndex, dataRow] of data.entries()) {
         const object: any = {};
-        for (const prop in schema.properties) {
-          if (schema.properties.hasOwnProperty(prop) && row[prop]) {
+        for (const propName in schema.properties) {
+          if (propName in dataRow) {
+            // We know that this is a ObjectSchemaProperty since we're reading it from the realm.schema
+            const propertySchema = schema.properties[
+              propName
+            ] as ObjectSchemaProperty;
+            const dataValue = dataRow[propName];
             try {
-              switch (schema.properties[prop]) {
-                case 'bool?':
-                  object[prop] = JSON.parse(row[prop].toLocaleLowerCase());
-                  break;
-                case 'int?':
-                  const intNumber = parseInt(row[prop], 10);
-                  if (isNaN(intNumber)) {
-                    throw new Error(
-                      `Can not parse ${row[prop]} as int at line ${lineIndex}`,
-                    );
-                  }
-                  object[prop] = intNumber;
-                  break;
-                case 'double?':
-                  const floatNumber = parseFloat(row[prop]);
-                  if (isNaN(floatNumber)) {
-                    throw new Error(
-                      `Can not parse ${row[prop]} as int at line ${lineIndex}`,
-                    );
-                  }
-                  object[prop] = floatNumber;
-                  break;
-                default:
-                  // string?
-                  object[prop] = row[prop];
+              if (propertySchema.optional && dataValue === '') {
```

Comment:
Comment: This seems wrong, or at least hard to actually insert empty strings on optional fields, but it seems this is a limitation in the CSV standard.

Reply:
Yeah, I agree. There is the edge case of empty strings. Ideally this should be a setting when importing.
Since its possible to go from `null` to an empty string via the UI (although thats 1 operation per row) but not the other way around, I think this is a good default for now (until we decide to invest more in this).